fe/fe-core/src/main/java/com/starrocks/sql/parser/TokenNumberListener.java

```diff
@@ -0,0 +1,23 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.parser;
+
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+public class TokenNumberListener extends StarRocksBaseListener {
+
+    private int maxTokensNum;
+
+    public TokenNumberListener(int maxTokensNum) {
+        this.maxTokensNum = maxTokensNum;
+    }
+
+    @Override
+    public void visitTerminal(TerminalNode node) {
+        int index = node.getSymbol().getTokenIndex();
+        if (index >= maxTokensNum) {
+            throw new OperationNotAllowedException("Sql exceeds %d tokens, please consider modify parse_tokens_limit variable.",
```

Comment:
Normal users don't understand tokens concept, we should make the error message to be user-friendly.

Reply:
have changed the error info more clear.